crates/bevy_ui_render/src/text2d.rs

```diff
@@ -0,0 +1,135 @@
+use bevy_text::{
+    ComputedTextBlock, PositionedGlyph, TextBounds,
+    TextColor, TextLayoutInfo,
+};
+use bevy_asset::Assets;
+use bevy_camera::visibility::ViewVisibility;
+use bevy_color::LinearRgba;
+use bevy_ecs::{
+    entity::Entity,
+    prelude::With,
+    system::{Commands, Query, Res, ResMut},
+};
+use bevy_image::prelude::*;
+use bevy_math::{Vec2};
+use bevy_render::sync_world::TemporaryRenderEntity;
+use bevy_render::Extract;
+use bevy_sprite::Anchor;
+use bevy_sprite_render::{ExtractedSlice, ExtractedSlices, ExtractedSprite, ExtractedSprites};
+use bevy_transform::prelude::GlobalTransform;
+use bevy_window::{PrimaryWindow, Window};
+
+/// This system extracts the sprites from the 2D text components and adds them to the
+/// "render world".
+pub fn extract_text2d_sprite(
+    mut commands: Commands,
+    mut extracted_sprites: ResMut<ExtractedSprites>,
+    mut extracted_slices: ResMut<ExtractedSlices>,
+    texture_atlases: Extract<Res<Assets<TextureAtlasLayout>>>,
+    windows: Extract<Query<&Window, With<PrimaryWindow>>>,
+    text2d_query: Extract<
+        Query<(
+            Entity,
+            &ViewVisibility,
+            &ComputedTextBlock,
+            &TextLayoutInfo,
+            &TextBounds,
+            &Anchor,
+            &GlobalTransform,
+        )>,
+    >,
+    text_colors: Extract<Query<&TextColor>>,
+) {
+    let mut start = extracted_slices.slices.len();
+    let mut end = start + 1;
+
+    // TODO: Support window-independent scaling: https://github.com/bevyengine/bevy/issues/5621
```

Comment:
Unrelated but I noticed that this PR is closed, so this TODO might no longer be relevant?

Reply:
i dont see how RenderLayers are relevant here, but I get the rest. Cant a text be drawn to multiple cameras regardless of render layer?